src/main/common/draft/draftMiddleware.ts

```diff
@@ -11,18 +13,19 @@ export class DraftMiddleware<T> {
     this.client = new DraftStoreClient<T>(draftType)
   }
 
-  retrieve (res: express.Response, next: express.NextFunction): void {
+  async retrieve (res: express.Response, next: express.NextFunction): Promise<void> {
     if (res.locals.isLoggedIn) {
-      this.client
-        .retrieve(res.locals.user.id, this.deserializeFn)
-        .then(draft => {
-          if (!draft) {
-            draft = this.deserializeFn(undefined)
-          }
-          res.locals.user[`${this.draftType}Draft`] = draft
-          next()
-        })
-        .catch(next)
+      try {
+        let draft: T = await this.client.retrieve(res.locals.user.id, this.deserializeFn)
+        if (!draft) {
+          draft = this.deserializeFn(undefined)
+        }
+        this.setUserData(draft, res.locals.user)
```

Comment:
As I understand that will always override persistence user data when case is retrieved, which gets potentially persisted next time save is made. That is a bit wired / unexpected behaviour for me. 

I would have liked to initiate draft metadata when draft is created for the first time and do not modify it afterwards.

Reply:
hmm, if we only initialise then we wouldn't get the data for old draft's when the user logs in

Reply:
I saw this issue but proposed solution looks like a big sacrifice to overcome the problem. Maybe we do not have to care about few existing drafts. Or maybe such metadata should be only saved at persist time and read as is, if does not exist then fine UI does not need it.

For me cleaner is to call `this.setUserData(draft, res.locals.user)` at save time so then owner is overridden with each save which makes it really a last owner / modifier.

Reply:
A `DraftStoreClient.save` would be the right place, the same as we set `draft.lastUpdateTimestamp`